src/XR/features/WebXRLightEstimation.ts

```diff
@@ -0,0 +1,366 @@
+
+    /**
+     * This observable will notify when the reflection cube map is updated.
+     */
+    public onReflectionCubeMapUpdatedObservable: Observable<BaseTexture> = new Observable();
+
+    /**
+    * Creates a new instance of the light estimation feature
+    * @param _xrSessionManager an instance of WebXRSessionManager
+    * @param options options to use when constructing this feature
+    */
+    constructor(
+        _xrSessionManager: WebXRSessionManager,
+        /**
+         * options to use when constructing this feature
+         */
+        public readonly options: IWebXRLightEstimationOptions
+    ) {
+        super(_xrSessionManager);
+        this.xrNativeFeatureName = "light-estimation";
+
+        if (this.options.createDirectionalLightSource) {
+            this.directionalLight = new DirectionalLight("light estimation directional", this._lightDirection, this._xrSessionManager.scene);
+            this.directionalLight.position = new Vector3(0, 8, 0);
+            // intensity will be set later
+            this.directionalLight.intensity = 0;
+        }
+
+        // https://immersive-web.github.io/lighting-estimation/
+        Tools.Warn("light-estimation is an experimental and unstable feature.");
+    }
+
+    /**
+     * While the estimated cube map is expected to update over time to better reflect the user's environment as they move around those changes are unlikely to happen with every XRFrame.
+     * Since creating and processing the cube map is potentially expensive, especially if mip maps are needed, you can listen to the onReflectionCubeMapUpdatedObservable to determine
+     * when it has been updated.
+     */
+    public get reflectionCubeMapTexture(): Nullable<BaseTexture> {
+        return this._reflectionCubeMap;
+    }
+
+    /**
+     * The most recent light estimate.  Available starting on the first frame where the device provides a light probe.
+     */
+    public get xrLightingEstimate(): Nullable<IWebXRLightEstimation> {
+        if (this._xrLightEstimate) {
+            return {
+                lightColor: this._lightColor,
+                lightDirection: this._lightDirection,
+                lightIntensity: this._intensity
+            };
+
+        }
+        return this._xrLightEstimate;
+    }
+
+    private _getCanvasContext(): WebGLRenderingContext | WebGL2RenderingContext {
+        if (this._canvasContext === null) {
+            this._canvasContext = this._xrSessionManager.scene.getEngine()._gl;
+        }
+        return this._canvasContext;
+    }
+
+    private _getXRGLBinding(): XRWebGLBinding {
+        if (this._xrWebGLBinding === null) {
+            let context = this._getCanvasContext();
+            this._xrWebGLBinding = new XRWebGLBinding(this._xrSessionManager.session, context);
+        }
+        return this._xrWebGLBinding;
+    }
+
+    /**
+     * Event Listener to for "reflectionchange" events.
+     */
+    private _updateReflectionCubeMap = (): void => {

```

Comment:
nit: Why not using `private _updateReflectionCubeMap(): void {` as for all other methods of the class?

Reply:
Good question! :-)
This is an event listener. to avoid binding this we force this to be the context using an arrow function